[Concept Entry] C# Interfaces (#3090)
Codecademy/docs: content/c-sharp/concepts/interfaces/interfaces.md

---
Title: 'Interfaces'
Description: 'An interface is an abstract contract that defines methods and properties that must be implemented by any class that implements the interface.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Interfaces'
  - 'Syntax'
  - 'Abstraction'
  - 'Documentation'
CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

An **interface** in C# is a contract that defines a set of methods, properties, events, and indexers that a class or struct must implement. Interfaces cannot be instantiated directly, but they can be implemented by classes and structs. They are used to achieve abstraction and polymorphism in C#.

### Syntax

> To create an interface in C#, you use the `interface` keyword. The syntax for defining an interface in C# is:

Reviewer: ```suggestion
An interface in C# is created using the `interface` keyword. The syntax for defining an interface in C# is:
```
No first- or second-person pronouns in Docs.

```
interface MyInterface
{
    void MyMethod();
    string MyProperty { get; set; }
    event EventHandler MyEvent;
}
```

To implement an interface in C#, you use the `:` symbol, followed by the name of the interface. The syntax for implementing an interface is as follows:

```
class MyClass : MyInterface
{
    public void MyMethod()
    {
        // implementation of MyMethod()
    }

    public string MyProperty
    {
        get { return "MyValue"; }
        set { }
    }

    public event EventHandler MyEvent;
}
```

### Example

> The following examples show various comment styles:

Reviewer: ```suggestion
The following example shows implementation of an interface:

```
